pyblnet/blnet.py

```diff
@@ -106,6 +106,51 @@ def _turn(self, digital_id, value, can_node=None):
         else:
             raise EnvironmentError('Can\'t set values with blnet web disabled')
 
+    def get_value(self,type='digital', ret='value', name=None, id=None, cached=None):
+        """
+          higher level interface to get a value or mode by name or id
+          ret: can be 'value', 'mode'
+          returns the value  if ret='value' or the mode if ret='mode' as first return value
+                  and the dictionary containing name, value, id and mode as a second return value
+          cached: in order to prevent polling data from BLNet with every call, 
+                  the data can be fetched once and stored and passed to this function
+
+
+        """
+        val=None
+        dic=None
+        if name is None and id is None: return val
+        if cached==None:
+          cached=self.fetch()
+        for key,v in cached[type].items():
+           if v['name']==str(name) or name==None: 
+             if v['id']==str(id) or id==None:
+               val=v[ret]
+               dic=v
+        return val, dic
+    
+    def get_digital_value(self ret='value', name=None, id=None, cached=None):
+        val, dic=self.get_value(type='digital', ret='value', name=name, id=id, cached=cached)
+        return val, dic
+    
+    def get_analog_value(self ret='value', name=None, id=None, cached=None):
+        val, dic=self.get_value(type='analog', ret='value', name=name, id=id, cached=cached)
+        return val, dic
+    
+    def get_energy_value(self ret='value', name=None, id=None, cached=None):
```

Comment:
Pretty sure there are commata missing after almost every self here.

Please do at least do a syntax check before merge requesting. I personally recommend setting up an IDE, it will spare you a lot of hustle and unnecessary errors.

Reply:
I am sorry. I did run this code on my linux machine and then copied it into the class (adding the self).